app/models/manager_refresh/inventory_collection_default/cloud_manager.rb

```diff
@@ -131,6 +131,17 @@ def orchestration_templates(extra_attributes = {})
       attributes.merge!(extra_attributes)
     end
 
+    def snapshots(extra_attributes = {})
+      attributes = {
+        :model_class                  => ::Snapshot,
+        :association                  => :snapshots,
+        :manager_ref                  => [:vm_or_template],
```

Comment:
@miha-plesko hm, can we have more snapshots per Vm or Image? If yes, the manager ref needs to have another column to uniquely identify them. Right now we are saying 1 snapshot per Vm/Image

Reply:
If I omit `:manager_ref` it means it'll be inventoried based on snapshot's `ems_ref`, right? We have unique `ems_ref` on snapshot, let me try out.

Reply:
right, if you omit it, the default is ```manager_ref => [:ems_ref]```

so might be that in this case we want ```manager_ref => [:vm_or_template, :ems_ref]```

or if the :ems_ref is unique under the whole provider, just ```:manager_ref => :ems_ref ``` should be enough

Reply:
to sum the choices:
1)
```manager_ref => [:ems_ref]``` which is the default, means :ems_ref is unique for the whole provider. Then we will just do ```.assign_attributes(:vm_or_template => parsed_vm)``` in the parser to link them together, which will make sure we first save the Vm then Snapshot

2) 
```manager_ref => [:vm_or_template, :ems_ref]``` means :ems_ref is unique under the Vm only
then with these we also have ```:parent_inventory_collections => [:vms, :miq_templates]``` which says the Snapshot is nested to Vm(or template). That usually means we fetch it from DB through Vms. And the main implication for targeted refresh is we always need to provide all Snapshots with the Vm target.